#!/usr/bin/env python3

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("globalEnable",    bool_t,    0, "A Boolean parameter", False)
gen.add("reboot_params",    bool_t,    0, "A Boolean parameter", False)


size_enum = gen.enum([ gen.const("empty_3x3",      str_t, "empty_3x3", "empty_3x3"),
                       gen.const("empty_10x10",     str_t, "empty_10x10", "empty_10x10"),
                       gen.const("hard_maze_10x10",      str_t, "hard_maze_10x10", "hard_maze_10x10"),
                       gen.const("random_10x10", str_t, "random_10x10", "random_10x10"),
                       gen.const("simple_maze_10x10", str_t, "simple_maze_10x10", "simple_maze_10x10"),
                       gen.const("test_10x10", str_t, "test_10x10", "test_10x10")],
                     "An enum to set size")


gen.add("map_name", str_t, 0, "A size parameter which is edited via an enum", "empty_10x10", edit_method=size_enum)
gen.add("botNumber",    int_t,    0, "An Integer parameter", 10,  1, 50)

size_enum2 = gen.enum([ gen.const("line",str_t, "line", "empty_3x3"),
                       gen.const("circle",str_t, "circle", "empty_10x10")],
                     "An enum to set size")
gen.add("distribution", str_t, 0, "A size parameter which is edited via an enum", "circle", edit_method=size_enum2)


gen.add("circle_center_x", double_t, 0, "A double parameter",    0, -5,   5)
gen.add("circle_center_y", double_t, 0, "A double parameter",    0, -5,   5)
gen.add("circle_radius", double_t, 0, "A double parameter",    .5, 0.5,   5)


gen.add("line_center_x", double_t, 0, "A double parameter",    0, -5,   5)
gen.add("line_center_y", double_t, 0, "A double parameter",    0, -5,   5)

size_enum3 = gen.enum([ gen.const("vertical",str_t, "vertical", "empty_3x3"),
                       gen.const("horizontal",str_t, "vertical", "empty_10x10")],
                     "An enum to set size")


gen.add("line_direction", str_t, 0, "A size parameter which is edited via an enum", "vertical", edit_method=size_enum3)
gen.add("line_separation", double_t, 0, "A double parameter",    0.5, 0.1,   2)

gen.add("BOID_RADIUS", double_t, 0, "A double parameter",    3, .1,   10)
gen.add("BOID_ANGLE", double_t, 0, "A double parameter",    160, 5,   360)
gen.add("BOID_MAX_VEL", double_t, 0, "A double parameter",    0.3, 0.05,   2)

gen.add("separation", double_t, 0, "A double parameter",    0.5, 0,   1)
gen.add("alignment", double_t, 0, "A double parameter",    0.5, 0,   1)
gen.add("cohesion", double_t, 0, "A double parameter",    2, 0,   1)



exit(gen.generate("reynold", "reynold", "dynamicVariables"))